### PR TITLE
support CREATE VIEW ... UnionStmt

### PR DESCRIPTION
**Pull request by XuHuaiyu**
### What problem does this PR solve? 
Before this:
``` sql
tidb> create view v as select * from t1 union select * from t1;
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use line 1 column 39 near "union select * from t1"
tidb> create view v as (select * from t1);
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use line 1 column 18 near "(select * from t1)"
```

### What is changed and how it works?
After this, create view on `union` and create view on `(select)` syntax are supported.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

N/A

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: parser.y [hunk 1]
```diff
@@ -3329,6 +3330,27 @@ CreateTableSelectOpt:
...
 
+CreateViewSelectOpt:
+	SelectStmt
+	{
+		$$ = $1.(*ast.SelectStmt)
+	}
+|
+	UnionStmt
+	{
+	    $$ = $1.(*ast.UnionStmt)
```
Comment (kennytm) [inline parser.y]:
```suggestion
		$$ = $1.(*ast.UnionStmt)
```

Indent using tabs

File: parser.y [hunk 2]
```diff
@@ -3349,10 +3371,15 @@ LikeTableWithOrWithoutParen:
...
+    "CREATE" OrReplace ViewAlgorithm ViewDefiner ViewSQLSecurity "VIEW" ViewName ViewFieldList "AS" CreateViewSelectOpt ViewCheckOption
     {
 		startOffset := parser.startOffset(&yyS[yypt-1])
-		selStmt := $10.(*ast.SelectStmt)
+		var selStmt ast.StmtNode
+		var ok bool
+		selStmt, ok = $10.(*ast.SelectStmt)
+		if !ok {
+			selStmt = $10.(*ast.UnionStmt)
+		}
```
Comment (kennytm) [inline parser.y]:
```suggestion
		selStmt := $10.(ast.StmtNode)
```

File: ast/ddl.go [hunk 3]
```diff
@@ -1247,7 +1247,10 @@ func (n *CreateViewStmt) Accept(v Visitor) (Node, bool) {
 	if !ok {
 		return n, false
 	}
-	n.Select = selnode.(*SelectStmt)
+	n.Select, ok = selnode.(*SelectStmt)
+	if !ok {
+		n.Select = selnode.(*UnionStmt)
+	}
```
[the remark this answers is not in the document]
Comment (kennytm) [reply ast/ddl.go]: This is still not resolved...

File: parser.y [hunk 4]
```diff
@@ -3349,10 +3371,11 @@ LikeTableWithOrWithoutParen:
  *          as select Col1,Col2 from table WITH LOCAL CHECK OPTION
  *******************************************************************/
 CreateViewStmt:
-    "CREATE" OrReplace ViewAlgorithm ViewDefiner ViewSQLSecurity "VIEW" ViewName ViewFieldList "AS" SelectStmt ViewCheckOption
+    "CREATE" OrReplace ViewAlgorithm ViewDefiner ViewSQLSecurity "VIEW" ViewName ViewFieldList "AS" CreateViewSelectOpt ViewCheckOption
     {
 		startOffset := parser.startOffset(&yyS[yypt-1])
-		selStmt := $10.(*ast.SelectStmt)
+		var selStmt ast.StmtNode
+		selStmt = $10.(ast.StmtNode)
```
Comment (kennytm) [inline parser.y]: You could use `:=` here...

Comment (XuHuaiyu) [reply parser.y]: Sorry, my fault

**Pull request merged**